### PR TITLE
CLN: Remove ipython 2.x compat

### LINKED ISSUE #25149: HTML repr benchmarks failing on Travis
e.g. https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=7904

```
2019-02-04T18:50:51.0110372Z [ 11.57%] ··· frame_methods.Repr.time_html_repr_trunc_mi                  failed
2019-02-04T18:50:51.0110695Z [ 11.57%] ···· Traceback (most recent call last):
2019-02-04T18:50:51.0111142Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/asv/benchmark.py", line 1039, in main_run_server
2019-02-04T18:50:51.0111204Z                    main_run(run_args)
2019-02-04T18:50:51.0111455Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/asv/benchmark.py", line 913, in main_run
2019-02-04T18:50:51.0111504Z                    result = benchmark.do_run()
2019-02-04T18:50:51.0111753Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/asv/benchmark.py", line 412, in do_run
2019-02-04T18:50:51.0111802Z                    return self.run(*self._current_params)
2019-02-04T18:50:51.0112042Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/asv/benchmark.py", line 506, in run
2019-02-04T18:50:51.0112105Z                    min_run_count=self.min_run_count)
2019-02-04T18:50:51.0112359Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/asv/benchmark.py", line 569, in benchmark_timing
2019-02-04T18:50:51.0112421Z                    timing = timer.timeit(number)
2019-02-04T18:50:51.0112648Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/timeit.py", line 176, in timeit
2019-02-04T18:50:51.0112695Z                    timing = self.inner(it, self.timer)
2019-02-04T18:50:51.0112884Z                  File "<timeit-src>", line 6, in inner
2019-02-04T18:50:51.0112934Z                  File "/home/vsts/work/1/s/asv_bench/benchmarks/frame_methods.py", line 226, in time_html_repr_trunc_mi
2019-02-04T18:50:51.0112977Z                    self.df3._repr_html_()
2019-02-04T18:50:51.0113024Z                  File "/home/vsts/work/1/s/pandas/core/frame.py", line 651, in _repr_html_
2019-02-04T18:50:51.0113065Z                    import IPython
2019-02-04T18:50:51.0113320Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/IPython/__init__.py", line 55, in <module>
2019-02-04T18:50:51.0113382Z                    from .terminal.embed import embed
2019-02-04T18:50:51.0113634Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/IPython/terminal/embed.py", line 16, in <module>
2019-02-04T18:50:51.0113686Z                    from IPython.terminal.interactiveshell import TerminalInteractiveShell
2019-02-04T18:50:51.0113952Z                  File "/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/IPython/terminal/interactiveshell.py", line 81, in <module>
2019-02-04T18:50:51.0114176Z                    if not _stream or not hasattr(_stream, 'isatty') or not _stream.isatty():
2019-02-04T18:50:51.0114223Z                ValueError: I/O operation on closed file
```

### PR DESCRIPTION
**Pull request by TomAugspurger**
Maybe closes https://github.com/pandas-dev/pandas/issues/25149

### CODE REVIEW

File: asv_bench/benchmarks/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+"""Pandas benchmarks."""
```
Comment (TomAugspurger) [inline asv_bench/benchmarks/__init__.py]: This is to ensure that ASV is run on the CI.

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -28,6 +28,24 @@ Other Enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Pandas 0.24.0 includes a number of API breaking changes.
```
Comment (jschendel) [inline doc/source/whatsnew/v0.25.0.rst]: 0.25.0 (and similar on line 33)

File: doc/source/whatsnew/v0.25.0.rst [hunk 3]
```diff
@@ -28,6 +28,24 @@ Other Enhancements
...
+Increased minimum versions for dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We have updated our minimum supported versions of dependencies (:issue:`25036`).
+If installed, we now require:
+
++-----------------+-----------------+----------+
+| Package         | Minimum Version | Required |
++=================+=================+==========+
+| IPython         | 3.0.0           |          |
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: is there a reason we don't just go to the py3 only version? (6.x I think)?

Comment (TomAugspurger) [reply doc/source/whatsnew/v0.25.0.rst]: No reason. This section can likely be remove entirely.

**Comment by jreback** [comment]
lgtm. did you want a whatsnew for this?

**Comment by qwhelan** [comment]
@TomAugspurger Agreed that this will fix the underlying issue. https://github.com/pandas-dev/pandas/pull/25237 can be reverted once this is merged.

**Comment by TomAugspurger** [comment]
No whatsnew since it’s impossible to run ipython 3 with pandas 0.25

**Pull request merged**